python/cuml/ts/stationarity.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+
+def _is_stationary(yi: np.ndarray, pval_threshold=0.05) -> bool:
+    """Single series stationarity test."""
+    ns = len(yi)
+
+    # Null hypothesis: data is stationary around a constant
+    e = yi - yi.mean()
+
+    # Table 1, Kwiatkowski 1992
+    crit_vals = [0.347, 0.463, 0.574, 0.739]
+    pvals = [0.10, 0.05, 0.025, 0.01]
+
+    s = np.cumsum(e)
+    # eq. 11
+    eta = s.dot(s) / ns**2
+
+    #########################
+    # compute eq. 10
+
+    # from Kwiatkowski et al. referencing Schwert (1989)
+    lags = int(np.ceil(12. * np.power(ns / 100., 1 / 4.)))
+
+    s2_A = e.dot(e)/ns
+
+    def w(_s, _l):
+        return 1-_s/(_l+1)
+
+    s2_B = 0.0
+    for j in range(1, lags+1):
+        eprod = np.dot(e[j:], e[:ns - j])
+        s2_B += 2/ns * w(j, lags) * eprod
+
+    s2 = s2_A + s2_B
+    #########################
+
+
+
+    kpss_stat = eta / s2
+    p_value = np.interp(kpss_stat, crit_vals, pvals)
+
+    # print("kpss_stat: {}, pvalue:{}".format(kpss_stat, p_value))
+
+    # higher p_value means higher chance data is stationary around constant.
+    return p_value > pval_threshold
+
+
+def stationarity(y: np.ndarray, pval_threshold=0.05) -> np.ndarray:
```

Comment:
Have filed issue #1207 to get this implemented as a separate prim inside c++. Let me know if you think that's not required.

Reply:
Seems like a good "starter" project for someone. I assume people will want to use this functionality, but I wasn't sure so I didn't want to invest too much time CUDAfying it.

Reply:
Fair enough. Makes perfect sense to start with python.